### ISSUE TITLE
Update cmake compatibility bounds

### ISSUE THREAD

**Issue by Luthaf** [chemfiles/chemfiles#388]
Since version 3.19, cmake is sending a warning about compatibility bounds for bzip2, netcdf, tng & molfiles external libraries:

```
Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.
```

We should update these bounds to explicitly indicate which cmake version we require. I think keeping support for 2.8.12 is good, so we could use their suggested upper bound instead.

**Comment by Luthaf** [comment]
After digging a bit, 2.8.12 is still supported, but the above packages have a minimal cmake version of 2.8.11. Given that the oldest cmake available in a LTS distro is 2.8.12 on CentOS 7, I think the fix is as simple as updating all bounds to 2.8.12.

**Comment by Luthaf** [comment]
Done in #402
